fitbenchmarking/controllers/tests/test_controllers.py

```diff
@@ -16,6 +16,9 @@
 from conftest import run_for_test_types
 from fitbenchmarking import test_files
 from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.controllers.controller_factory import \
+        ControllerFactory
+from fitbenchmarking.controllers.nlopt_controller import nlopt
```

Comment:
This is another change from pylint complaining. It was saying that things were being used later in the file that may or may not be imported. That is why, I updated the tests to use ControllerFactory to create the controllers. I think the `TEST_TYPE` will still provide protection against running tests on software that is not installed.
 
<img width="406" alt="Screenshot 2024-08-13 at 14 02 04" src="https://github.com/user-attachments/assets/0b9611d6-00ae-4d5e-b346-fd9e41302de4">

I could not get rid of the nlopt import because it is being used later on for status checks. 

<img width="378" alt="Screenshot 2024-08-13 at 14 31 44" src="https://github.com/user-attachments/assets/8f799c05-60e6-43c8-b991-0c2c3db8ce90">

Reply:
Yeah, my only worry was that it wasn't imported previously for the matlab tests. I guess it's a default (pip installable) one so it's no big issue. I'm also wondering if we should just import nlopt rather than importing it through the controller (while we're fixing things?)

Reply:
I will make that change now. I want to look into nlopt in this issue #1051 later on. I can't install nlopt in any new environment on the latest M1 and M3 macs. Right now, nlopt is installed for the default tests but when we add the mac tests it may not be installable on `macos-latest`. We might need to use the `macos-13` runner instead.

Reply:
Ok, the mac issue is going to be a problem no matter how we write this test. So I think it's good to keep that for it's own PR.